src/SimpleGraphs/generators/staticgraphs.jl

```diff
@@ -372,3 +372,44 @@ function CliqueGraph(k::T, n::T) where {T <: Integer}
     add_edge!(g, 1, (n - 1) * k + 1)
     return g
 end
+
+"""
+    LadderGraph(n)
+
+Create a graph consisting of `2n` nodes and `3n-2` edges.
+
+### Implementation Notes
+Preserves the eltype of `n`. Will error if the required number of vertices
+exceeds the eltype.
+"""
+function LadderGraph(n::T) where {T <: Integer}
+    n <= 0 && SimpleGraph{T}(0)
+    n == 1 && PathGraph(T(2))
+
+    g = SimpleGraph{T}(2*n)
+    for i in 1:(n-1)
+        add_edge!(g, i, i+1)
+        add_edge!(g, n+i, n+i+1)
+        add_edge!(g, i, n+i)
+    end
+    add_edge!(g, n, 2*n)
+    return g
+end
+
+"""
+    CircularLadderGraph(n)
+
+Create a graph consisting of `2n` nodes and `3n` edges.
+
+### Implementation Notes
+Preserves the eltype of the partitions vector. Will error if the required number of vertices
```

Comment:
Nothing to do with partitions :)
Maybe mention that this is equal to a prim graph `Y_n` so that people who search the documentation for that will find it.